File: androidApp/src/main/java/com/teewhydope/app/ui/character/mapper/PersonDetailPresentationToUiModelMapper.kt
package com.teewhydope.app.ui.character.mapper

import com.teewhydope.app.common.mapper.PresentationToUiMapper
import com.teewhydope.app.presentation.character.viewmodel.PersonDetailViewState
import com.teewhydope.app.ui.character.model.PersonDetailUiModel


class PersonDetailPresentationToUiModelMapper :
    PresentationToUiMapper<PersonDetailViewState.Loaded, PersonDetailUiModel>() {
    override fun map(input: PersonDetailViewState.Loaded) = PersonDetailUiModel(
        id = input.personDetail.id,
        name = input.personDetail.name,
        height = input.personDetail.height,
        mass = input.personDetail.mass,
        image = input.personDetail.image,
        birthYear = input.personDetail.birthYear,
        gender = input.personDetail.gender,
        homeWorld = input.personDetail.homeWorld,
        url = input.personDetail.url,
    )
}

File: iosApp/iosApp/People/Mapper/PersonDetailPresentationToUiModelMapper.swift
//
//  PersonDetailPresentationToUiModelMapper.swift
//  iosApp
//
//  Created by ibrahim animasahun on 09/04/2025.
//  Copyright © 2025 orgName. All rights reserved.

import Foundation
import CommonData

class PersonDetailPresentationToUiModelMapper: PresentationToUiMapper{
    typealias T = PersonDetailViewStateLoaded
    
    typealias U = PersonDetailUiModel
    
    
    func toUi(input: PersonDetailViewStateLoaded) -> PersonDetailUiModel {
        return .init(
            id: Int(input.personDetail.id),
            name: input.personDetail.name,
            height: input.personDetail.height,
            mass: input.personDetail.mass,
            image: input.personDetail.image,
            birthYear: input.personDetail.birthYear,
            gender: input.personDetail.gender,
            homeWorld: input.personDetail.homeWorld,
            url: input.personDetail.url,
        )
    }
    
}


